### ISSUE TITLE
Icon Request: TypeScript

### ISSUE THREAD

**Issue by magnus-trent** [FortAwesome/Font-Awesome#11890]
It would be nice to have an icon for TypeScript, Microsoft's superset of Javascript. General usage I couldn't say much past the fact that I personally use the icon on information pages when talking about what tools I use for projects.

Thank you.

**Comment by lundyio** [comment]
bump

**Comment by DevidCIC** [comment]
The addition of TypeScript would be very nice.

**Comment by blyndusk** [comment]
Yeah, TypeScript is very used actually, it would be very nice to add it !

**Comment by AdamBarnwell** [comment]
+1

**Comment by adhiravishankar** [comment]
+1

**Comment by OliveCh12** [comment]
Typescript icon still not updated ?

**Comment by alevasseurEG** [comment]
+1

**Comment by Xelorion** [comment]
+1

**Comment by emanuelescarabattoli** [comment]
+1

**Comment by emanuelescarabattoli** [comment]
+1

**Comment by admarko** [comment]
+1

**Comment by theusoliveira** [comment]
Please add this icon! I really need it!

**Comment by blyndusk** [comment]
After more than 2 years, no response from the Font Awesome team for this icon, and the dev icons in general. 

So, I'm using [Devicon](https://konpa.github.io/devicon/) from @kompa.

**Comment by UnlockDep** [comment]
LETS GET THIS TO THE TOP OF THE LEADER BOARD

**Comment by ariel-frischer** [comment]
Bump

**Comment by Weh-gh** [comment]
I agree yes adding a TypeScript icon would be great!

**Comment by lbirkert** [comment]
+1

**Comment by nolway** [comment]
+1

**Comment by devanup** [comment]
Why is nobody showing love to Typescript icon? Is it that ugly?

**Comment by tagliala** [comment]
Hi @devanup 

You may be interested in taking a look at our [Requesting Icons FAQs](https://fontawesome.com/support#requesting-icons), they contain a lot of information about how requests work

**Comment by zmunro** [comment]
@tagliala Why dont you take a look at the thread and see that everyone wants this icon added.

**Comment by ShadiestGoat** [comment]
Bump - I don't understand why this isn't already a thing

**Comment by joeykamsteeg** [comment]
+1

**Comment by ManojTGN** [comment]
+1

**Comment by mariandeimel** [comment]
+1

**Comment by tagliala** [comment]
@trulysinclair hi, I'm going to reopen this request because of the huge number of upvotes

I want to clarify that commenting with "+1" is against our contributing guidelines and requests should be upvoted via reactions on the opening post

If you don't want to receive updates, please use the "Unsubscribe" feature on the right sidebar, because otherwise all upvotes in this request would be lost and it should restart from scratch

<img width="358" alt="image" src="https://github.com/FortAwesome/Font-Awesome/assets/556268/bd93992c-f8a2-4900-889d-d9daada6dcbd">

**Comment by cutpix** [comment]
We need it!

**Comment by FeatureSpitter** [comment]
How many years are needed to add this logo?

**Comment by khanhowe** [comment]
bump

**Comment by sphynxlee** [comment]
Today, TypeScript is still unavailable.

**Comment by manthanank** [comment]
No typescript icon available

**Comment by davedawkins** [comment]
Yes please! Pro subscriber here

**Comment by jasstsg** [comment]
Commenting in case it also helps bump the priority--since I see both comments and reactions to the initial issue post are tracked here: https://fontawesome.com/leaderboard/brands

**Comment by zmunro** [comment]
The maintainers of this repo truly dont give a shit

**Comment by ThiaudioTT** [comment]
still we don't have it boiiiss

**Comment by AnubisXHyperX** [comment]
7 Years, and still no icon ?

**Comment by davedawkins** [comment]
I got you. Some one make a PR for this please
![image](https://github.com/user-attachments/assets/2e7250e6-da9a-4704-9b19-ee5a53fba50f)

**Comment by naterfute** [comment]
Typescript logo when?

**Comment by TomerMiz10** [comment]
wow, still no TypeScript logo icon for Font-Awesome. A 7-year-old request.

**Comment by naterfute** [comment]
Literally never going to happen

**Comment by rfig81** [comment]
+1

**Comment by thfrod** [comment]
2025 and still no ts logo :/

**Comment by naterfute** [comment]
Mfw still waiting for the logo 🥹

**Comment by oardi** [comment]
+1

**Comment by RuMont** [comment]
7 years to create a TS icon means they definitely wont do it. Unsubscribing from this issue.
I drop this here in case somebody needs it:

https://www.svgrepo.com/vectors/typescript/

**Comment by MohammadAliJa** [comment]
> 7 years to create a TS icon means they definitely wont do it. Unsubscribing from this issue. I drop this here in case somebody needs it:
> 
> https://www.svgrepo.com/vectors/typescript/

I think they need an extra 7 years to be completed, 😕.

**Comment by DJFriar** [comment]
Adding another comment to hopefully bump this request up. Especially odd it is missing considering their documentation even includes instructions on how to support TypeScript.

**Comment by kuraykaraaslan** [comment]
Hi Font Awesome team,

It’s been over 13 years since TypeScript became a mainstream language and yet there is still no official Font Awesome icon for it. TypeScript is now a first-class language across both frontend and backend ecosystems — supported directly by major frameworks like React, Angular, Vue, and even Node.js itself.

To help move this forward, here is a fully functional and styled icon definition based on the existing visual identity of TypeScript:

`
  const customTSIcon: IconDefinition = {
    prefix: "fab",
    iconName: "typescript",
    icon: [
      32,
      32,
      [],
      "f0c8",
      "M0 16v16h32v-32h-32zM25.786 14.724c0.813 0.203 1.432 0.568 2.005 1.156 0.292 0.312 0.729 0.885 0.766 1.026 0.010 0.042-1.38 0.974-2.224 1.495-0.031 0.021-0.156-0.109-0.292-0.313-0.411-0.599-0.844-0.859-1.505-0.906-0.969-0.063-1.594 0.443-1.589 1.292-0.005 0.208 0.042 0.417 0.135 0.599 0.214 0.443 0.615 0.708 1.854 1.245 2.292 0.984 3.271 1.635 3.88 2.557 0.682 1.031 0.833 2.677 0.375 3.906-0.51 1.328-1.771 2.234-3.542 2.531-0.547 0.099-1.849 0.083-2.438-0.026-1.286-0.229-2.505-0.865-3.255-1.698-0.297-0.323-0.87-1.172-0.833-1.229 0.016-0.021 0.146-0.104 0.292-0.188s0.682-0.396 1.188-0.688l0.922-0.536 0.193 0.286c0.271 0.411 0.859 0.974 1.214 1.161 1.021 0.542 2.422 0.464 3.115-0.156 0.281-0.234 0.438-0.594 0.417-0.958 0-0.37-0.047-0.536-0.24-0.813-0.25-0.354-0.755-0.656-2.198-1.281-1.651-0.714-2.365-1.151-3.010-1.854-0.406-0.464-0.708-1.010-0.88-1.599-0.12-0.453-0.151-1.589-0.057-2.042 0.339-1.599 1.547-2.708 3.281-3.036 0.563-0.109 1.875-0.068 2.427 0.068zM18.276 16.063l0.010 1.307h-4.167v11.839h-2.948v-11.839h-4.161v-1.281c0-0.714 0.016-1.307 0.036-1.323 0.016-0.021 2.547-0.031 5.62-0.026l5.594 0.016z"
    ],
  };
`

**Comment by nolway** [comment]
Maybe one day...

**Comment by omarhurani** [comment]
can we at least get a reason on why this isn't being done?

**Comment by lai-bianca** [comment]
It would be very useful to have this icon! I see that this was self-assigned recently, so I hope that means that this is actually going to be added.

**Comment by sensibleworld** [comment]
So... our backlog of brand requests has reached a point where even we can't pretend we're keeping up. Some of these issues were opened before I had gray hair. And I've had gray hair for a LONG time. On top of that, treating requests as issues has always made it way too hard to see the forest for the trees.

To fix this, we are now managing all brand requests in **GitHub Discussions**. No more +1s or thumbs-ups needed. There's an upvote system built-in.

We are closing existing brand requests as part of this transition, but **we are not saying no** to your request.

**If this brand request is still relevant for you, please re-open it in Discussions** (and feel free to include a link back to this issue):
-> https://github.com/FortAwesome/Font-Awesome/discussions/new?category=brand-requests

And listen. We know this is annoying. If you need to vent, here are some pre-written responses you can use when opening the new discussion:

* Cool cool cool. No worry at all. I love re-requesting a brand that's redesigned three times while this issue was open.
* If an issue closes after seven years and nobody responds, did it ever really exist?
* I see you have chosen violence today. Re-opening per instructions.
* What's next? I have to file a TPS report, too?
* I am filled with the fury of a thousand suns. Why do I have to re-open the discussion? Couldn't you have just bulk converted all of the issues to discussions? What? You couldn't? What do you mean Github removed that feature earlier this year? You'd have to manually click the "convert to discussion" button on thousands of issues, while also ensuring it gets added to the correct category? That's a bunch of bull-phooey.
* No biggie.

**Comment by sensibleworld** [comment]
Hey folks! This is in progress. It'll be in our next release.

**Comment by kuraykaraaslan** [comment]
I literally added the code above… just copy and paste this. I mean, how hard can it be?

**Comment by jabronus** [comment]
> I literally added the code above… just copy and paste this. I mean, how hard can it be?

As Jory mentioned, this is already ready for our next release 🙂
